demo/client/backend/src/fabricClient.js

```diff
@@ -0,0 +1,477 @@
+/*
+* Copyright IBM Corp All Rights Reserved
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+'use strict';
+
+const FabricCAClient = require('fabric-ca-client');
+const { Gateway, Client, FileSystemWallet, X509WalletMixin } = require('fabric-network');
+const fs = require('fs');
+const path = require('path');
+
+////////////////////////////  Global constants  /////////////////
+//  errorcodes
+const SUCCESS = 0;
+const FAILURE  = 1;
+const USER_EXISTS = 2;
+const USER_UNAUTHORIZED = 3;
+
+/////////////////////////////  input  //////////////////////////////////////////
+const orgName        = 'Org1MSP';
+const orgDomainName  = 'org1.example.com';
+//const userName       = 'Admin';  //  hardcoded !
+const org1Affiliation = 'org1';  //  different from MSP;  used in CA
+
+//  affiliation is used in CA;  to add to the list of affiliation, customize
+//  fabric-ca-server-config.yaml and pass in as cafiles in start command.
+//  If not, use 'org1' and 'org2' - case sensitive, which is defined in
+// the default  fabric-ca-server-config.yaml.
+
+/////////////////////////////  end of input  ///////////////////////////
+
+//  global variables loaded initially
+var configdata;
+var mspDir;
+var connectionProfile;
+var wallet;
+var adminUserName;
+var secret;
+var bLocalHost;  //  used in setUserContext
+
+
+//  other global variables
+var strResponse;
+var gateway;
+var network;
+var contract;
+
+//  exported functions
+async function connectToNetwork(channelName, chaincodeID) {
+    console.log (">>> In function, connectToNetwork: channelName, chaincodeID = ", channelName, chaincodeID);
+
+    await beginNetworkConnect();
+
+    try {
+      gateway = await new Gateway();
+    }  catch (error) {
+      console.log(`Error creating new gateway`);
+      console.log(error.stack);
+    }
+
+    await gateway.connect(connectionProfile,
+        	  {
+              wallet,
+              identity:            adminUserName,
+              discovery:           { enabled: false, localhost: bLocalHost },
+              eventHandlerOptions: { strategy: null }
+      		  });
+    console.log ("Connected to gateway");
+    network =  await gateway.getNetwork (channelName);
+    console.log ("Connected to channel, " + channelName);
+
+    contract =  await network.getContract (chaincodeID);
+    console.log ("Connected to chaincode, ", chaincodeID);
+    console.log ('...........................');
```

Comment:
`fabricClient.js` is very verbose. In general logging is a good thing but here I wish to reduce it or introduce the concept of log levels? Seems that most log statements were primarily used for developing this code but don't bring necessarily any value when running the demo?!

Please revisit the logging!